platform-tests/src/test/java/org/junit/platform/engine/support/hierarchical/ResourceLockTests.java

```diff
@@ -12,68 +12,120 @@
 
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.platform.engine.support.hierarchical.ExclusiveResource.GLOBAL_READ;
+import static org.junit.platform.engine.support.hierarchical.ExclusiveResource.GLOBAL_READ_WRITE;
 
 import java.util.List;
 import java.util.concurrent.locks.ReentrantLock;
 
 import org.junit.jupiter.api.Test;
+import org.junit.platform.engine.support.hierarchical.ExclusiveResource.LockMode;
 
-@SuppressWarnings("resource")
 class ResourceLockTests {
 
 	@Test
 	void nopLocksAreCompatibleWithEverything() {
 		var nopLock = NopLock.INSTANCE;
 
+		assertTrue(nopLock.isCompatible(nopLock));
 		assertTrue(nopLock.isCompatible(NopLock.INSTANCE));
-		assertTrue(nopLock.isCompatible(new SingleLock(anyReentrantLock())));
-		assertTrue(nopLock.isCompatible(new SingleLock.GlobalReadLock(anyReentrantLock())));
-		assertTrue(nopLock.isCompatible(new SingleLock.GlobalReadWriteLock(anyReentrantLock())));
-		assertTrue(nopLock.isCompatible(new CompositeLock(List.of(anyReentrantLock()))));
+		assertTrue(nopLock.isCompatible(new SingleLock(anyResource(), anyReentrantLock())));
```

Comment:
🤔 introduce helper methods like you already did for `resource` to further reduce the noise

```
SingleLock singleLock(ExclusiveResource er) {
    return new SingleLock(er, anyReentrantLock());
}

CompositeLock compositeLock(List<ExclusiveResource> ers) {
    return new CompositeLock(er, ers.stream().map(__ -> anyReentrantLock()).toList());
}
```

You could also introduce assertion methods.
```
void compatible(ResourceLock a, ResourceLock b) {
    assertTrue(a.isCompatible(b));
}

void incompatible(ResourceLock a, ResourceLock b) {
    assertFalse(a.isCompatible(b));
}
```

Then this line would be

```
compatible(nopLock, singleLock(anyResource()));
```

Reply:
Polished in f18f048196d82ad98b43ad96a6cbcfca436f63f3.